evalml/tests/automl_tests/test_automl.py

```diff
@@ -2251,6 +2257,79 @@ def test_automl_pipeline_random_seed(mock_fit, mock_score, random_seed, X_y_mult
             assert automl.get_pipeline(row['id']).random_seed == random_seed
 
 
+@pytest.mark.parametrize("ensembling", [True, False])
+@pytest.mark.parametrize("ensemble_split_size", [0.1, 0.2])
+@patch('evalml.automl.automl_search.split_data')
+@patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.3})
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+def test_automl_ensembling_training(mock_fit, mock_score, mock_split_data, ensemble_split_size, ensembling, X_y_binary):
+    X, y = X_y_binary
+    X_train, X_ensemble, y_train, y_ensemble = split_data(X, y, problem_type='binary', test_size=ensemble_split_size)
+    mock_split_data.return_value = (X_train, X_ensemble, y_train, y_ensemble)
+    # don't train the best pipeline since we check usage of the ensembling CV through the .fit mock
+    ensemble_pipelines = len(get_estimators("binary")) + 2
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', random_state=0, n_jobs=1, max_batches=ensemble_pipelines, ensembling=ensembling,
+                          train_best_pipeline=False, optimize_thresholds=False, _ensembling_split_size=ensemble_split_size)
+    automl.search()
+    if ensembling:
+        assert automl.ensembling
+        # check that the X_train data is all used for the length
+        assert len(X_train) == (len(mock_fit.call_args_list[-2][0][0]) + len(mock_score.call_args_list[-2][0][0]))
```

Comment:
Trying to follow along, but why are we checking this against the length of the mock_fit and length of the mock_score args? 🤔

Reply:
This checks that the second-to-last pipeline (which isn't ensembling) uses the full X_train dataset between the fit and score calls. The `mock_fit.call_args_list[-2][0][0]` gets the X input for the fit call, and the same for `mock_score`. I just did this to make sure that we're using the training data we expect as opposed to the ensembling data. 

Hopefully that clarifies!